tests/utils/managed_utils/client.rs

```diff
@@ -104,7 +104,20 @@ impl Client {
         let group_state = group_states
             .get_mut(&group_id)
             .ok_or(ClientError::NoMatchingGroup)?;
-        group_state.process_message(message.clone())?;
+        // Prevent feeding further messages to client after it was removed
+        // by one of the messages.
+        if !group_state.is_active() {
+            return Ok(());
```

Comment:
Don't you want to return an error message when the group is no longer active?

Reply:
Come to think about it, this check should no longer be necessary. After removing it, the group will throw an error when it's inactive and we're calling `process_message`. And you're right, that's how it should be.